base/channels.jl

```diff
@@ -246,9 +246,10 @@ Append an item `v` to the channel `c`. Blocks if the channel is full.
 For unbuffered channels, blocks until a [`take!`](@ref) is performed by a different
 task.
 """
-function put!(c::Channel, v)
+function put!(c::Channel{T}, v) where T
     check_channel_state(c)
-    isbuffered(c) ? put_buffered(c,v) : put_unbuffered(c,v)
+    vT = convert(T, v)
```

Comment:
It's no longer a performance issue to reuse the name `v` for this. Style is a different question, but I thought I'd mention it.

Reply:
Thanks. I was partly skeptical when I did this. Will add a commit to reuse name.